amida-tech/indaba-client
# INBA 711 Offset and a bunch of PropType checks.

Calculates an offset for question number if you have sections.

#### What does this PR do?
If you have sections, now numbers the questions appropriately.

#### Related JIRA tickets:
https://jira.amida-tech.com/browse/INBA-711

#### How should this be manually tested?
Create a survey with two, preferably three or four sections. Give each section a varying number of questions. 

Assign a user to it then go to the task review page. When you choose a different section, you should see the question numbers match as though you were seeing all the questions at once. So if section 2 contains questions 4 and 5, they should be labeled as such. See the screenshot for an idea. 

Also added a bunch of propType checks. You don't have to worry about those unless you see an error. 

#### Background/Context

#### Screenshots (if appropriate):
![image](https://user-images.githubusercontent.com/4026454/35901647-b99d4780-0ba5-11e8-90cc-bf0954983fda.png)
